drivers/Watchdog.h

```diff
@@ -28,11 +28,9 @@
 #include <cstdio>
 
 namespace mbed {
-/** \ingroup mbed-os-public */
```

Comment:
Why did we drop the public?

Reply:
It is still in public. See line 33.

Reply:
yeah but why change of groups? We used to have the
Public -> [Drivers, RTOS, ...]
Private -> [...]
is this change affecting the group tree?

Reply:
The generated Doxygen ouptut has not changed. This change simply makes the Doxygen comment of this file in line with other files under `drivers`.